### PR TITLE
docs(expect): APIResponse assertions

### LINKED ISSUE #10189: [Feature] expect(response).toBeOk() to print http response status and body
### Discussed in https://github.com/microsoft/playwright/discussions/10175

<div type='discussions-op-text'>

<sup>Originally posted by **AndyGBC** November  9, 2021</sup>
Hi all, below is a code snippet that will use the APIContextRequest to make a post request and create a company.  This works fine, though I am sure someone could help improve this, if so, feel free :)

My question is, if this fails and returns a 403 status, the expect on the status will obviously catch it and fail, but how can I go about getting more information about why it failed?

By this I mean, if after the POST request has been made, I can console.log the response.json() which would show me some details about the failure but that won't appear in the HTML report which is what I would like to see if possible.

Any help appreciated.

`async createCompany(name: string, shortName: string, description: string): Promise<string> {

        expect(name.trim().length != 0).toBeTruthy();
        
        const response = await this.baseAPI.context.post(`https://some.endpoint/create`, { 
            headers:{
                'X-BC-JWT': this.baseAPI.api.getToken()
            },
            data: {
                kind: 'company',
                name: name,
                short_name: shortName,
                descr: description,
            }
        });

        expect(response.status()).toEqual(200);
        expect(await response.json()).toHaveProperty('artifact');

        let companyUri: string = '';

        await response.json().then(data => {
            companyUri = data.artifact;
        });

        expect(companyUri.length != 0).toBeTruthy();
        return companyUri;
    }`
</div>

### PR DESCRIPTION
**Pull request by yury-s**
Fixes #10189

### CODE REVIEW

**Review by aslushnikov — APPROVED** [review]

File: docs/src/api/class-apiresponseassertions.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+- returns: <[APIResponseAssertions]>
+
+Makes the assertion check for the opposite condition. For example, this code tests that the response status is not successfull:
+
+```js
+expect(response).not.toBeOK();
+```
+
+```java
+assertThat(page).not().isOK();
```
Comment (ltsuda) [inline docs/src/api/class-apiresponseassertions.md:63]: Should this supposed to assert `response`?

Comment (yury-s) [reply docs/src/api/class-apiresponseassertions.md:63]: yep

File: docs/src/api/class-apiresponseassertions.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# class: APIResponseAssertions
+* langs: js
```
Comment (mxschmitt) [inline docs/src/api/class-apiresponseassertions.md:2]: shall we ship it to the languages too?

File: docs/src/api/class-apiresponseassertions.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+```js
+expect(response).not.toBeOK();
+```
+
+```java
+assertThat(response).not().isOK();
+```
+
+## method: APIResponseAssertions.toBeOK
```
Comment (mxschmitt) [inline docs/src/api/class-apiresponseassertions.md:66]: not_to_be_ok seems missing for Python.

File: docs/src/api/class-apiresponseassertions.md (hunk 2 above)
Comment (yury-s) [reply docs/src/api/class-apiresponseassertions.md:2]: dunno, let's discuss tomorrow

**Pull request merged**